### PR TITLE
an action help language CTRL-] in tag.c fixed

### PR DESCRIPTION
**Pull request by Tatsuki-I**
When I set translated help(for example: `set helplang=ja,en` in .vimrc), open translated help and press `CTRL-]`, Vim opens help in translated help, but open English help and press `CTRL-]`, Vim opens translated help.
I'd like vim to open same language help as current opened help. 
So, I fixed tag.c.

### CODE REVIEW

**Comment by vim-ml** [comment]
Hi Tatsuki,

2016-11-12(Sat) 16:49:47 UTC+9 Tatsuki:

> When I set translated help(for example: set helplang=ja,en in .vimrc), open translated help and press CTRL-], Vim opens help in translated help, but open English help and press CTRL-], Vim opens translated help.
> 
> I'd like vim to open same language help as current opened help.
> 
> So, I fixed tag.c.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/vim/vim/pull/1249
> 
> Commit Summary
> 
>   an action help language CTRL-] in tag.c fixed
> 
> File Changes
> 
> ```
> M
> src/tag.c
> (29)
> ```
> 
> Patch Links:
> 
>   https://github.com/vim/vim/pull/1249.patch
>   https://github.com/vim/vim/pull/1249.diff

Nice patch!
It is indeed written in the document.
:h helplang

> ```
>    When using CTRL-] and ":help!" in a non-English help file Vim will
>    try to find the tag in the current language before using this option.
> ```

I wrote a patch to add a test for this.  (Attached)
Check it out.

BTW, I found a bit wrong indentation.
tag.c;1527+             /\* Prefer help tags according to 'helplang'.  Put the

TAB SPC SPC SPC SPC TAB SPC SPC SPC SPC
It should be:

## TAB TAB SPC SPC SPC SPC

Best regards,
Hirohito Higashi (a.k.a. h_east)

File: src/tag.c [hunk 1]
```diff
@@ -1476,6 +1476,16 @@ find_tags(
      * When the tag file is case-fold sorted, it is either one or the other.
      * Only ignore case when TAG_NOIC not used or 'ignorecase' set.
      */
+#ifdef FEAT_MULTI_LANG
+    /* Set a flag if the file extension is .txt */
+    int is_txt = 0;
+      if ((flags & TAG_KEEP_LANG)
```
Comment (mattn) [inline src/tag.c]: fix indentation

File: src/tag.c [hunk 2]
```diff
@@ -1476,6 +1476,16 @@ find_tags(
      * When the tag file is case-fold sorted, it is either one or the other.
      * Only ignore case when TAG_NOIC not used or 'ignorecase' set.
      */
+#ifdef FEAT_MULTI_LANG
+    /* Set a flag if the file extension is .txt */
+    int is_txt = 0;
```
Comment (mattn) [inline src/tag.c]: there is a statment before this declaration. vim doesn't support C99. so move up this declaration with #ifdef FEAT_MULTI_LANG.

**Comment by brammool** [comment]
The declaration
 int is_txt = FALSE;
Needs to go at the start of the block.

File: src/tag.c [hunk 3]
```diff
@@ -1356,6 +1356,7 @@ find_tags(
     char_u	*help_lang_find = NULL;		/* lang to be found */
     char_u	help_lang[3];			/* lang of current tags file */
     char_u	*saved_pat = NULL;		/* copy of pat[] */
+    int 	is_txt = FALSE;			/* flag of file extension */
```
Comment (h-east) [inline src/tag.c:1359]: You need to enclose this in #ifdef FEAT_MULTI_LANG ～ #endif

Comment (h-east) [reply src/tag.c:1359]: Ah, It was already surrounded. sorry.

File: src/tag.c (hunk 1 above)
Comment (Tatsuki-I) [reply src/tag.c]: Thank you.
I fixed it.

File: src/tag.c (hunk 2 above)
Comment (Tatsuki-I) [reply src/tag.c]: I moved the definition to 1359.

**Comment by Tatsuki-I** [comment]
this test was created by h_east.
I'm sorry to forget to wrote about it.

File: src/tag.c (hunk 3 above)
Comment (Tatsuki-I) [reply src/tag.c:1359]: It is not necessary that you apologize
Thank you for teaching me.

**Comment by mattn** [comment]
@Tatsuki-I thanks. looks good to me.

**Pull request closed without merging**